content/agent/prometheus.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Prometheus Check
+kind: documentation
+---
+
+## Overview
+
+This page details how to collect metrics and events from a Prometheus data source by writing an Prometheus Check, a Python plug-in to the Datadog Agent. This page looks first at the `PrometheusCheck` interface, and then propose a simple Prometheus Check that collects timing metrics and status events from [Kube DNS](https://github.com/DataDog/integrations-core/blob/master/kube_dns/check.py).
+
+## Setup
+
+First off, ensure you've properly installed the [Agent](http://app.datadoghq.com/account/settings#agent) on your machine. If you run into any issues during the setup, [contact our support](/help).
+
+## Prometheus check interface 
+
+`PrometheusCheck` is [a mother class](https://github.com/DataDog/dd-agent/blob/master/checks/prometheus_check.py) providing a structure and some helpers
+to collect metrics, events and service checks exposed via Prometheus, minimal configuration for checks based on this class include:
+
+- Overriding self.NAMESPACE
+- Overriding self.metrics_mapper
+- Implementing the check method
+AND/OR
+- Create method named after the prometheus metric they will handle (see self.prometheus_metric_name)
+
+## Your first Prometheus check
+### Configuration
+
+<div class="alert alert-warning">
+The names of the configuration and check files must match. If your check
+is called <code>mycheck.py</code> your configuration file <em>must</em> be
+named <code>mycheck.yaml</code>.
+</div>
+
+Configuration for a Prometheus Check is almost the same as a regular Agent Check, please refer to the [dedicated Agent check documentation to learn more](/agent/agent_checks/#configuration)
+
+The main difference is to include the variable `prometheus_endpoint` in your `check.yaml` file. This goes into `conf.d/kube_dns.yaml`:
+
+```yaml
+init_config:
+
+instances:
+    # url of the metrics endpoint of prometheus
+  - prometheus_endpoint: http://localhost:10055/metrics
+    [{}]
+```
+
+### Writing your Prometheus check
+All Prometheus checks inherit from the `PrometheusCheck` class found in `checks/prometheus_check.py`: 
+
+```python
+from checks.prometheus_check import PrometheusCheck
+
+class KubeDNSCheck(PrometheusCheck):
+```
+
+#### Overriding `self.NAMESPACE`
+
+`NAMESPACE` is the prefix metrics will have. It needs to be hardcoded in your check class:
+
+```python
+from checks.prometheus_check import PrometheusCheck
+
+class KubeDNSCheck(PrometheusCheck):
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        super(KubeDNSCheck, self).__init__(name, init_config, agentConfig, instances)
+        self.NAMESPACE = 'kubedns'
+```
+
+#### Overriding `self.metrics_mapper`
+
+`metrics_mapper` is a dictionary where the keys are the metrics to capture and the values are the corresponding metrics names to have in datadog.
```

Comment:
`metrics_mapper` is a dictionary where the key is the metric to capture and the value is the corresponding metric name in Datadog.

Reply:
👍